### PR TITLE
Important - CIN Portfolio Upgrades

### PR DESCRIPTION
**Pull request by v-stamenova**
# MAJOR CHANGES READ CAREFULLY
The following pull request introduces big changes to the current working environment 

## Docker changes
The pull request introduces some updates connected to the docker containerisation of the whole application structure. Starting with the most relevant:
### 1. Containerisation of the SvelteKit client
Since the client is now also within a container, `cd ./frontend` and `npm run dev` are NO longer needed.
The only thing needed currently is `docker compose up`

**Since now the client is also within the docker network, you cannot access the API Gateway with `localhost:3011`. Now when you interact with the API Gateway you need to use the container itself: `apigateway:3011`**

### 2. Removing Unnecessary images
In the previous structure, there was a pattern that within each microservice there was a `Dockerfile` that in a matter of fact was the same in each microservice - therefore for each microservice an image was created that was the same but with different names. To reduce code duplication a folder called `common` is created in the backend directory that contains the `Dockerfile`.

## ESLint
ESLint checker is installed in each microservice and the sveltekit client. Most of the rules I used are taken from [Frans](https://github.com/HZ-HBO-ICT/ts-skeleton-app/blob/main/.eslintrc.cjs). I added a `.vscode` folder that contains some configuration for the ESLint checker extension. Feel free to modify the rules, add new ones, remove others, just please don't put ignores to reduce work, use ignores responsibly. 

_ESLint has a svelte extension that managed to fix all of the frontend issues by itself so less work for us there - more merge conflicts though!_

**Don't forget to get the VSCode ESLint extension installed on your WSL! If you want to run the linting tests you can run: `npx eslint [src/file]`**

The sveltekit and the users microservice are fixed so - the api gateway, moods and habits must be fixed.

## Github Workflows
Since the configuration of the project got updated the workflows should be updated as well.
### Build Workflow
The build workflow just uses a docker action and checks if the containers manage to build properly
### ESLint Workflow
The ESLint checker runs currently in the frontend and in the users microservice. When we fix the issues in the other microservices they need to be added within that `.yml`. 
### Unit testing workflow
Basically the same as the last time `backend.yml`

### CODE REVIEW

**Review by LauraBirau — APPROVED** [review]
Amazing!
